package Package1.Generics;

import java.util.*;

class SumNumbers {
    public static double sumNumbers(List<? extends Number> list) {
        double sum = 0;
        for (Number n : list) {
            sum += n.doubleValue();
        }
        return sum;
    }

    public static void main(String[] args) {
        List<Integer> ints = Arrays.asList(1, 2, 3);
        List<Double> doubles = Arrays.asList(1.5, 2.5);

        System.out.println(sumNumbers(ints));
        System.out.println(sumNumbers(doubles));
    }
}
